machine-check/src/index.ts

```diff
@@ -1,2 +1,42 @@
-export { proc } from './traverse.js'
-export { toARSM, getSubscriptions } from './arsm.js'
+import { check_swarm, check_projection } from '../pkg/machine_check.js'
+
+export type Protocol<Label> = {
+  initial: string
+  transitions: { source: string; target: string; label: Label }[]
+}
+export type SwarmLabel = {
+  cmd: string
+  logType: string[]
+  role: string
+}
+
+export type MachineLabel =
+  | { tag: 'Execute'; cmd: string; logType: string[] }
+  | { tag: 'Input'; eventType: string }
+
+export type SwarmProtocolType = Protocol<SwarmLabel>
+export type MachineType = Protocol<MachineLabel>
+
+export type Subscriptions = Record<string, string[]>
+
+export type Result = { type: 'OK' } | { type: 'ERROR'; errors: string[] }
+
+export function checkSwarmProtocol(proto: SwarmProtocolType, subscriptions: Subscriptions): Result {
+  const p = JSON.stringify(proto)
+  const s = JSON.stringify(subscriptions)
+  const result = check_swarm(p, s)
+  return JSON.parse(result)
```

Comment:
The types between Rust and JS don't "match"? I thought that there would be a way to export a dict or similar

Reply:
AFAIK, langs that are compiled to WASM does not have 1-to-1 data structure mapping with JS, data that goes from WASM-world to the JS-world and vice versa has to be serialized (as bytes, or JSON string). WASM-ecosystem provides JSValue, but I don't think it matters much compared to serde_json.

It is in a way the similar to C<-->Rust FFI under the hood